### ISSUE TITLE
WriteRecord is not changing lines after 3.x update

### ISSUE THREAD

**Issue by pierre-weceipt** [JoshClose/CsvHelper#780]
Hi, I tried to update CsvHelper to 3.1 but `WriteRecord` method is not changing line at each call. Is it intended?
I've got no issues with successive call of `WriteRecords` though.

Thanks

**Comment by JoshClose** [comment]
Yes, that's intended. You need to call `NextRecord` afterwards. This is so you can use `WriteRecord` and still write more if you need to. I need to document this.

**Comment by pierre-weceipt** [comment]
Now I understand! Maybe we could have some config available to be able to au change line without adding a `NewRecord` at each `WriteRecord` call. Both usecase would then be available.

**Comment by JoshClose** [comment]
@pierre-weceipt What are you doing besides `WriteRecord` in your loop?

**Comment by pierre-weceipt** [comment]
@JoshClose some other stuff not linked with the csv process. I get no benefits from having NextRecord separated from WritzRecord in this case.

**Comment by JoshClose** [comment]
If you use `WriteRecords` you can pass in a list of all your records and you don't have to worry about it. If you want to loop the records manually, you'll need to call `NextRecord` on each loop.

**Comment by pierre-weceipt** [comment]
I'm using WriteRecords too but here I can't. I need the loop to deserialise json string into an object that is then used to write the record.

**Comment by JoshClose** [comment]
I don't see a good reason to have a configuration to have it automatically do `NextRecord`. You'd still have an extra line of code. Are you more concerned with people not realizing that's what they need to do? It's documented, but I could also mention it in the XML comments so it shows up in intellisense.

**Comment by pierre-weceipt** [comment]
Well in the non default cases you would have to add a line of code. But maybe most of people don't need the auto mode and then you don't need to change anything.
But if (like me) you use same config in multiple places then you only need to set it once ^^. 
I had no issue with finding the info in the doc, adding it to intellisense would be nice though.

**Comment by JoshClose** [comment]
I'll add it to the XML comments too.
